utilities/write_batch_with_index/write_batch_with_index.cc

```diff
@@ -539,13 +539,15 @@ void WriteBatchWithIndex::MultiGetFromBatchAndDB(
   // after the transaction finishes.
   for (size_t i = 0; i < num_keys; ++i) {
     MergeContext merge_context;
-    PinnableSlice* pinnable_val = &values[i];
-    std::string& batch_value = *pinnable_val->GetSelf();
+    std::string batch_value;
     Status* s = &statuses[i];
+    PinnableSlice* pinnable_val = &values[i];
```

Comment:
Should we expect the user input values are all not pinned? I think it should be user's responsibility to reset the value if needed.

Reply:
Is it the user's responsibility to guarantee that all values are not pinned?  The db methods that return a PinnableSlice say nothing about that property.

Reply:
Without break the compatibility, I guess it's hard to restrict that now.